### ISSUE TITLE
During the merge, does this plugin do a rebase?

### ISSUE THREAD

**Issue by mshah-nisum-com** [jenkinsci/ghprb-plugin#624]
I want to know if the Github Pull request merger post task does a merge or a rebase.

**Comment by Ippo343** [comment]
Disclaimer: I'm not affiliated with the project, so this answer might be wrong.

I dug through the code and it seems that the merge is done on [this line](https://github.com/jenkinsci/ghprb-plugin/blob/master/src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java#L213).
The [merge method](https://github.com/kohsuke/github-api/blob/master/src/main/java/org/kohsuke/github/GHPullRequest.java#L356-L358) is implemented in [kohsuke/github-api](https://github.com/kohsuke/github-api/), which ends up calling [this method](https://github.com/kohsuke/github-api/blob/master/src/main/java/org/kohsuke/github/GHPullRequest.java#L384-L390) with `method = null`, so I believe it defaults to a normal merge.

**TL;DR:** it does a normal merge.

**Comment by bjoernhaeuser** [comment]
@Ippo343 thats correct, thank you for the clarification.
